### DISCUSSION TITLE
Error compiling LISF-LDT

### DISCUSSION THREAD

**Discussion by laurizioA** [NASA-LIS/LISF#1433] [Q&A]
Hello!

I'm trying to install LISF and it keeps giving me an error.

I installed ESMF 8.5 in 'usr/local/esmf', but when indicating the installation paths in 'export LDT_MODESMF=/usr/local/esmfmodO/Linux.gfortran.64.mpiuni.default' and 'export LDT_LIBESMF=/ usr/local/esmf/lib/libO/Linux.gfortran.64.mpiuni.default' the following error occurs:

-------------------------------------------------- -----------------
Compiling LDT

[INFO] Compiling LDT source code

gfortran -I/usr/include/ -I/usr/include/ -I/usr/include/ -I/usr/include/ -I/usr/include/ -I/usr/include/ -I/usr/include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../core/LDT_constantsMod.F90
../core/LDT_constantsMod.F90:24:7:

     use ESMF, only : ESMF_MAXPATHLEN
        1
Fatal Error: Can't open module file ‘esmf.mod’ for reading at (1): No such file or directory
compilation terminated.
Makefile:108: recipe for target 'LDT_constantsMod.o' failed
make: *** [LDT_constantsMod.o] Error 1

[ERR] Compile failed
-------------------------------------------------- -----------------

The 'esmf.mod' module is located within $LDT_MODESMF.

I emphasize that I also carried out 'export LD_LIBRARY_PATH=$LDT_HDF5/lib:$LDT_LIBESMF:$LDT_NETCDF/lib:$LDT_ECCODES/lib:$LDT_ECCODES/lib:$LDT_HDF4/lib:$LDT_HDFEOS/lib:$LDT_GDAL/lib:$LD_LIBRARY_PATH' .

Has anyone ever had a similar error?

**Comment by dmocko** [comment]
HI @laurizioA 

You said above that you defined LDT_MODESMF as: `export LDT_MODESMF=/usr/local/esmfmodO/Linux.gfortran.64.mpiuni.default`

However, I believe that it should be set as: `export LDT_MODESMF=/usr/local/esmf/mod/modO/Linux.gfortran.64.mpiuni.default`

Can you please try again?  Go into LISF/ldt/make, run "gmake realclean", and then re-configure and re-compile.

If that doesn't work, please attach your configure.ldt file.
